Add JSON-LD schema generation for testimonial cards

roperev opened the pull request:
- Implemented functions to sanitize text and generate JSON-LD schema for testimonials.
- Added logic to extract review details, including review text, author name, product name, rating, and publication date.
- Injected the generated schema into the document head, ensuring no duplicates are created.
- Enhanced error handling for schema injection to prevent page breaks on failure.

You can see a validation of the schema here: https://validator.schema.org/#url=https%3A%2F%2F76-schematestimonial--idfc--aemsites.aem.page%2Fcredit-card%2Frupay-credit-card

Fix #76

Schema output looks like this in `head`:
<img width="844" height="1012" alt="image" src="https://github.com/user-attachments/assets/fbf8c86f-4d07-4ac6-a90f-0bb8a847c287" />


Test URLs:
- Before: https://main--idfc--aemsites.aem.page/credit-card/rupay-credit-card
- After: https://76-schematestimonial--idfc--aemsites.aem.page/credit-card/rupay-credit-card

Commit:
Add JSON-LD schema generation for testimonial cards

- Implemented functions to sanitize text and generate JSON-LD schema for testimonials.
- Added logic to extract review details, including review text, author name, product name, rating, and publication date.
- Injected the generated schema into the document head, ensuring no duplicates are created.
- Enhanced error handling for schema injection to prevent page breaks on failure.

helms-charity commented on blocks/cards/cards.js:
```diff
@@ -1,6 +1,200 @@
+    '@context': 'https://schema.org',
+    '@type': 'Product',
+    name: 'IDFC FIRST Bank RuPay Credit Card',
+    description: 'Apply for RuPay Credit Card at IDFC FIRST Bank & get 3X rewards on UPI transactions. Enjoy exclusive benefits, cashback, seamless digital payments.',
```
Get this description from the page meta description instead.

helms-charity commented on blocks/cards/cards.js:
```diff
@@ -1,6 +1,200 @@
+  return {
+    '@context': 'https://schema.org',
+    '@type': 'Product',
+    name: 'IDFC FIRST Bank RuPay Credit Card',
```
Get the page title from the metadata for this (for now).

helms-charity commented on blocks/cards/cards.js:
```diff
@@ -1,6 +1,200 @@
+    });
+    reviewText = sanitizeText(reviewText);
+
+    // Extract author name from h5
```
These H5/H6/P U queries are ok for now, but I made a refactor ticket for later https://github.com/aemsites/idfc/issues/133 . We shouldn't have the author as an H5 because it is breaking the semantic HTML hierarchy (there's no H3 or H4 in between the H2 and this H5).

Commit:
Enhance JSON-LD schema generation for testimonial cards

- Added dynamic metadata extraction for page title, description, and Open Graph properties.
- Improved product schema by incorporating optional fields such as URL, image, published and modified dates, and category.
- Updated product name and brand name extraction logic to handle variations in the page title.
